code/game/machinery/dna_infuser/dna_infusion.dm

```diff
@@ -0,0 +1,76 @@
+
+///returns a boolean whether a human can be infused
+/atom/movable/proc/can_infuse(mob/feedback_target)
+	if(!isliving(src))
+		if(feedback_target)
+			feedback_target.balloon_alert(feedback_target, "no dna!")
+		return FALSE
+	if(!ishuman(src))
+		if(feedback_target)
+			feedback_target.balloon_alert(feedback_target, "dna too simple!")
+		return FALSE
+	var/mob/living/carbon/human/human_infusing = src
+	// Checked by can_mutate but a special message is good
+	if(HAS_TRAIT(human_infusing, TRAIT_BADDNA))
+		if(feedback_target)
+			feedback_target.balloon_alert(feedback_target, "dna is corrupted!")
+		return FALSE
+	if(!human_infusing.can_mutate())
+		if(feedback_target)
+			feedback_target.balloon_alert(feedback_target, "dna is missing!")
+		return FALSE
+	return TRUE
```

Comment:
Rather than `isliving(src)` `ishuman(src)`, make use of OOP
```dm
/atom/movable/proc/can_infuse(mob/feedback_target)
 	if(feedback_target)
		feedback_target.balloon_alert(feedback_target, "no dna!")
	return FALSE

/mob/living/can_infuse(mob/feedback_target)
 	if(feedback_target)
		feedback_target.balloon_alert(feedback_target, "dna too simple!")
	return FALSE

/mob/living/carbon/human/can_infuse(mob/feedback_target)
	...
```